numpy/lib/function_base.py

```diff
@@ -1801,6 +1801,10 @@ def place(arr, mask, vals):
            [44, 55, 44]])
 
     """
+    if not isinstance(arr, np.ndarray):
```

Comment:
It feels little more awkward in this case because ```_insert``` is not an ```ndarray``` method.  I suppose you could do the following:
```
try:
   put = arr.put
except AttributeError:
   raise TypeError(...)
return _insert(arr, mask, vals)
```
Though it seems like more contrived compared to the others since we're relying on a method that has nothing to do with what is being called.